examples/rust/battleship/src/main.rs

```diff
@@ -12,6 +12,348 @@
+    println!("How many players today? (0|1|2)");
+    let mut number = String::new();
+    stdin().read_line(&mut number).expect("Failed to read line");
+    let count_players: u32 = number.trim().parse().expect("Not a number");
+    if count_players > 2 {
+        println!("Too many players...");
+    } else if count_players == 0 {
+        run_auto_bot();
+    } else {
+        println!("number of players, {}", count_players);
+        let mut player1 = setup_player_1();
+        let mut player2 = setup_player_2();
+
+        let player1_init = player1.battleship.init().unwrap();
+        let player2_init = player2.battleship.init().unwrap();
+
+        player1.battleship.on_init_msg(&player2_init).unwrap();
+
+        player2.battleship.on_init_msg(&player1_init).unwrap();
+
+        for n in 1..10 {
+            if n % 2 == 0 {
+                println!(
+                    "{} turn, what are your attack coordiantes? (i.e.: 0,0)",
+                    player1.name
+                );
+                let mut attack_coord1 = String::new();
+                stdin()
+                    .read_line(&mut attack_coord1)
+                    .expect("Failed to read line");
+                let vec: Vec<&str> = attack_coord1.split(",").collect();
+
+                let j = round(
+                    &mut player1.battleship,
+                    &mut player2.battleship,
+                    vec[0].trim().parse::<u32>().unwrap(),
+                    vec[1].trim().parse::<u32>().unwrap(),
+                );
+                println!("It's a {:?}", j);
+            } else {
+                println!(
+                    "{} turn, what are your attack coordiantes? (i.e.: 0,0)",
+                    player2.name
+                );
+                let mut attack_coord2 = String::new();
+                stdin()
+                    .read_line(&mut attack_coord2)
+                    .expect("Failed to read line");
+                let vec: Vec<&str> = attack_coord2.split(",").collect();
+
+                let k = round(
+                    &mut player2.battleship,
+                    &mut player1.battleship,
+                    vec[0].trim().parse::<u32>().unwrap(),
+                    vec[1].trim().parse::<u32>().unwrap(),
+                );
+                println!("It's a {:?}", k);
+            }
+        }
+    }
+}
+
+fn setup_player_1() -> Players {
+    println!("What's player's 1 name?");
+    let mut player1_string = String::new();
+    stdin()
+        .read_line(&mut player1_string)
+        .expect("Failed to read line");
+
+    let player1_state = GameState {
+        ships: [
+            setup_carrier(),
+            setup_battleship(),
+            setup_cruiser(),
+            setup_submarine(),
+            setup_destroyer(),
+        ],
+        salt: 0xCAFECAFE,
+    };
+
+    return Players {
+        name: player1_string,
+        battleship: Battleship::new(player1_state),
+    };
+}
+
+fn setup_player_2() -> Players {
```

Comment:
Is there any difference between `setup_player_1` and `setup_player_2`? If there is anything, use parameters so that you can have single `setup_player` function that can be called twice.

Reply:
not sure what i was thinking, but totally agree, last commit should address this